enc/encdb.c

```diff
@@ -17,7 +17,7 @@
 #define ENC_DEFINE(name) rb_encdb_declare(name)
 #define ENC_SET_BASE(name, orig) rb_enc_set_base((name), (orig))
 #define ENC_SET_DUMMY(name, orig) rb_enc_set_dummy(name)
-#define ENC_DUMMY_UNICODE(name) rb_encdb_set_unicode(rb_enc_set_dummy(ENC_REPLICATE((name), name "BE")))
```

Comment:
Removing `rb_encdb_set_unicode()` is necessary here, so that:
```
> "\u3042".encode("UTF-32BE").force_encoding("UTF-32")
=> "\x00\x00\x30\x42"
> "\u3042".encode("UTF-32BE").force_encoding("ISO-2022-JP")
=> "\x00\x00\x30\x42"
```
and not:
```
s = "\u3042".encode("UTF-32BE")
s.dup.force_encoding("UTF-32").inspect
=> "\"\\u0000\\u00000B\""
irb(main):005:0> s.dup.force_encoding("UTF-16").inspect
=> "\"\\u0000\\u00000B\""
```
Since the latter seems quite wrong (interpreting single bytes as unicode/ASCII characters in UTF-16/32)